library/ssl_tls12_server.c

```diff
@@ -3233,6 +3241,12 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
          */
         if( md_alg != MBEDTLS_MD_NONE )
         {
+            if( dig_signed == NULL && dig_signed_len > 0)
```

Comment:
The calculation of `dig_signed_len` has undefined behavior if `dig_signed` is null, so it's pointless to check this here.

It might be a good idea to check that `dig_signed != NULL`, since the fact that it's the case requires some high-level knowledge (`mbedtls_ssl_ciphersuite_uses_server_signature( ciphersuite_info )` implies `mbedtls_ssl_ciphersuite_uses_ecdhe( ciphersuite_info ) || mbedtls_ssl_ciphersuite_uses_dhe( ciphersuite_info )` — easy to see with the definitions of these functions though — and `defined(MBEDTLS_KEY_EXCHANGE_WITH_SERVER_SIGNATURE_ENABLED)` implies `defined(MBEDTLS_KEY_EXCHANGE_SOME_DHE_ENABLED) || defined(MBEDTLS_KEY_EXCHANGE_SOME_ECDHE_ENABLED)`). But if so it needs to be done before using `dig_signed`.